pkg/controller/account/account_controller.go

```diff
@@ -373,7 +373,7 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		if err != nil {
 			var returnErr error
 			if aerr, ok := err.(awserr.Error); ok {
-				reqLogger.Error(returnErr, fmt.Sprintf("Unable to create AWS connection with SRE credentials, AWS Error Message: %s", aerr.Message()))
+				reqLogger.Error(returnErr, fmt.Sprintf("Unable to create AWS connection with SRE credentials, AWS Error Code: %s, AWS Error Message: %s", aerr.Code(), aerr.Message()))
```

Comment:
You addressed the log msgs, by adding Code and additional info to them, but I understand another point of this card is to handle specific AWS Errors (like we do in checkcaseResolution function), instead of just checking if err != nil for success.

Reply:
@rogbas, I talked to @jharrington22 and @fahlmant and they indicated that it would be an ongoing process since we need to see possible errors in order to efficiently build cases for them. If you have insight on errors we know of I can definitely start adding those.

Reply:
ok

Reply:
If the messages and codes are going to be followed up on later is it worth adding a keyword to make extracting them from the rest of the log easier? Maybe the message body already well known and unique enough. Maybe this operator log isn't very chatty? Just thinking aloud.

Reply:
The main concern of this card is that we were not reporting the AWS error code and message when we get an error from the AWS API. This makes debugging and handing those errors in the future hard if we don't know what they are.

@c-e-brumm The errors will include the account name which is unique, so if we have an issue with a particular account we'll be able to extract the messages fairly easily. Good thought though.

Reply:
Its hard to predict what errors we'll want to handle for each service so logging them will allow us to easily handle them in the future.

Reply:
Having clarified that it's going to be a gradual process of matching to specific AWS Errors, I see other places where the error msg could be improved to add aws Code. Everytime we call a function from the `awsclient.Client` interface, examples: AttachAdminUserPolicy, CreateUserAccessKey.

Reply:
Hopefully @ALimaRedHat will catch them all, i've asked that we look at every AWS API call and ensure we're bubbling up the code and message. This is still a WIP but feel free to help point them out.

Reply:
@jharrington22 probably not part of that card, but we could start addressing some of the errors related to aws rate limiting us

Reply:
@rogbas :100: - thats pretty much the intent of this card, lets see which are real errors and where we are being rate limited.